File: common_algorithms/linear_search/java/linearSearch.java
class LinearSearch
{
    public static Boolean linearSearch(int key, int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            if(arr[i] == key)
            {
                return true;
            }
        }
        return false;
    }


    public static void main(String[] args)
    {
        int[] arr = {1, 2, 3, 4, 5, 6, 8, 12, 234, 234235, 2342354, 23465465, 234242345};

        // These should return True
        System.out.println(linearSearch(4, arr));
        System.out.println(linearSearch(6, arr));
        System.out.println(linearSearch(8, arr));
        System.out.println(linearSearch(234, arr));
        System.out.println(linearSearch(23465465, arr));

        // These should return False
        System.out.println(linearSearch(0, arr));
        System.out.println(linearSearch(7, arr));
        System.out.println(linearSearch(22, arr));
    }
}

File: common_algorithms/linear_search/python/linearSearch.py
def linearSearch(key, arr):
    #This is a linear search algorithm determining if the given value exists
    # It does not return the index althought the big O notation complexity value would be the same
    for i in range(len(arr)):
        if arr[i] == key:
            return True
    return False


if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5, 6, 8, 12, 234, 234235, 234342354, 23423546546745, 2342423456434574]

    print(linearSearch(7, arr)) #False
    print(linearSearch(1, arr)) #True
    print(linearSearch(2, arr)) #True
    print(linearSearch(234, arr)) #True
    print(linearSearch(2342354654674522, arr)) #False